workflow/rules/stats.smk

```diff
@@ -0,0 +1,30 @@
+rule samtools_flagstat:
+    input:
+        "results/{step}/{samples_units}.bam"
+    output:
+        "results/{step,[^./]+}/{samples_units}.{step}.flagstat"
+    wrapper:
+        "0.60.0/bio/samtools/flagstat"
+
+rule samtools_idxstats:
+    input:
+        bam = "results/{step}/{samples_units}.bam",
+        idx = "results/{step}/{samples_units}.bam.bai"
+    output:
+        "results/{step,[^./]+}/{samples_units}.{step}.idxstats"
+    log:
+        "logs/{step,[^./]+}/{samples_units}.{step}.log"
```

Comment:
The regex to restrict a wildcard only has an effect when specified in `output`, so it should be omitted in `log` fields to avoid confusion on where to set it. Same below in `rule samtools_stats`.